pyanaconda/modules/storage/checker/utils.py

```diff
@@ -225,6 +227,52 @@ def verify_gpt_biosboot(storage, constraints, report_error, report_warning):
                                "'biosboot' type partition."))
 
 
+def verify_petitboot_compatibility(storage, constraints, report_error, report_warning):
+    """ Verify the petitboot version compatibility.
+
+    :param storage: a storage to check
+    :param constraints: a dictionary of constraints
+    :param report_error: a function for error reporting
+    :param report_warning: a function for warning reporting
+    """
+    if arch.get_arch() == "ppc64le" and os.access("/sys/firmware/opal", os.F_OK):
+        # Check the version of the petitboot loader.
+        version = _get_petitboot_version()
```

Comment:
I don't think we care about the petitboot version at all. We need to ensure that the firmware kernel version (for OPAL systems) is 5.10 or newer. The kernel version can be found in `/proc/device-tree/ibm,firmware-versions/linux` or by executing `lsmcode` - see bug [1977236](https://bugzilla.redhat.com/show_bug.cgi?id=1977236).
@sharkcz , can you please confirm?

Reply:
- @jstodola is right, it's kernel version in the OPAL firmware that matters
- I would call the function/method `verify_opal_compatibility`
- the check for arch and `/sys/firmware/opal` could be switched to `if get_platform() == 'PowerNV'`
- if you can't positively detect the kernel version, then I would assume it's "bad/old"

Reply:
Thanks for the pointers! I have got confused by https://bugzilla.redhat.com/show_bug.cgi?id=1997832#c2. Hopefully, it is fine now.